packages/esm-service-queues-app/src/patient-search/visit-form/existing-visit-form.component.tsx

```diff
@@ -0,0 +1,119 @@
+import React, { useCallback, useState } from 'react';
+import { Button, ButtonSet, Form, Row, Stack } from '@carbon/react';
+import { useTranslation } from 'react-i18next';
+import {
+  type ConfigObject,
+  ExtensionSlot,
+  showSnackbar,
+  useConfig,
+  useLayoutType,
+  type Visit,
+} from '@openmrs/esm-framework';
+import { addQueueEntry, useVisitQueueEntries } from '../../active-visits/active-visits-table.resource';
+import { type SearchTypes } from '../../types';
+import styles from './visit-form.scss';
+
+interface ExistingVisitFormProps {
+  toggleSearchType: (searchMode: SearchTypes, patientUuid) => void;
+  visit: Visit;
+  closePanel: () => void;
+}
+
+const ExistingVisitForm: React.FC<ExistingVisitFormProps> = ({ visit, toggleSearchType, closePanel }) => {
+  const { t } = useTranslation();
+  const isTablet = useLayoutType() === 'tablet';
+  const [isSubmitting, setIsSubmitting] = useState(false);
+
+  const config = useConfig() as ConfigObject;
+  const visitQueueNumberAttributeUuid = config.visitQueueNumberAttributeUuid;
+  const { mutate } = useVisitQueueEntries('', '');
+
+  if (!visit) {
+    return null;
+  }
+
+  const handleSubmit = useCallback(
```

Comment:
Don't really need useCallback here (see [this](https://react.dev/reference/react/useCallback#should-you-add-usecallback-everywhere)). It makes it difficult to maintain with the list of dependencies.